sdc/datatypes/hpat_pandas_series_rolling_functions.py

```diff
@@ -307,6 +305,32 @@ def apply_minp(arr, ddof, minp):
     gen_hpat_pandas_series_rolling_ddof_impl(arr_var))
 
 
+@sdc_register_jitable
+def pop_mean(value, nfinite, result):
+    """Calculate the window mean without old value."""
+    if numpy.isfinite(value):
+        nfinite -= 1
+        if nfinite:
+            result = ((nfinite + 1) * result - value) / nfinite
+        else:
+            result = numpy.nan
+
+    return nfinite, result
+
+
+@sdc_register_jitable
+def put_mean(value, nfinite, result):
+    """Calculate the window mean with new value."""
+    if numpy.isfinite(value):
+        nfinite += 1
+        if numpy.isnan(result):
+            result = value / nfinite
+        else:
+            result = ((nfinite - 1) * result + value) / nfinite
```

Comment:
Probably we can introduce new function something like `get_result`. Which for `mean` would be:
```python
def get_mean(nfinite, result):
    if nfinite:
        return result/nfinite
    return numpy.nan
```

And then, call it inside `result_or_nan`:
```python
@sdc_register_jitable
def result_or_nan(get, nfinite, minp, result):
    """Get result taking into account min periods."""
    if nfinite < minp:
        return numpy.nan

    return get(nfinite, result)
```

For `mean` we can do recalculations on `put`/`pop` stage, but for more complex functions (std, var) it is not that easy.

Hope, it wouldn't ruin our performance